spec/services/google/parse_service_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Google::ParseService, type: :service do
+  describe '#call' do
+    it 'retrieves non-ads entries', vcr: 'google/ruby_11_non_ads' do
+      html = Google::SearchService.new.search! 'ruby'
+
+      non_ads = described_class.new.call(html).search_entries.count { |e| e.kind == 'non_ads' }
+
+      expect(non_ads).to eq(11)
+    end
+
+    it 'extracts the non-ads entry URLs', vcr: 'google/ruby_11_non_ads' do
+      html = Google::SearchService.new.search! 'ruby'
+
+      entry = described_class.new.call(html).search_entries.find do |e|
+        e.kind == 'non_ads' && e.urls.any? { |u| /ruby-lang.org/ =~ u }
+      end
+
+      expect(entry).not_to be_nil
+    end
+
+    it 'counts the number of links', vcr: 'google/monitor_50_links' do
+      html = Google::SearchService.new.search! 'monitor'
+
+      links_count = described_class.new.call(html).links_count
+
+      expect(links_count).to eq(50)
+    end
+
+    it 'includes the html of the page', vcr: 'google/monitor_50_links' do
+      html = Google::SearchService.new.search! 'monitor'
+
+      result_page = described_class.new.call(html).result_page_html
+
+      expect(result_page).to be_a String
```

Comment:
should we test the String here is not an empty string `''` 😄

Reply:
Fixed in 9db72f6cbabd519c0a1d76cf3c1a887383cf9bf2